### PR TITLE
ci(docs): simplify and add kuma-cp.md config docs

### PR DESCRIPTION
**Pull request by lahabana**
- kuma-cp.md auto generated from kuma-cp.defaults.yaml to be more easily embedded in the docs
- rewrite make to make it more straightforward and work with -j

### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues --
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s) --
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [x] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

**Comment by lahabana** [comment]
@mergifyio backport release-2.2 release-2.1 release-2.0 release-1.8 release-1.7

File: mk/docs.mk [hunk 1]
```diff
@@ -1,30 +1,34 @@
...
+docs/generated/resources: ## Generate Mesh API reference
+	mkdir -p $@ && $(PROTOC) \
+		--kumadoc_out=$@ \
 		--plugin=protoc-gen-kumadoc=$(PROTOC_GEN_KUMADOC) \
-		$(PROTOS)
+		$(DOCS_PROTOS)
+
+.PHONY: docs/generated/kuma-cp.md
+docs/generated/kuma-cp.md: ## Generate Mesh API reference
+	@mkdir -p $(dir $@)
```
Comment (bartsmykla) [inline mk/docs.mk:29]: Just a nit
```suggestion
	@mkdir -p $(@D)
```

File: mk/docs.mk [hunk 2]
```diff
@@ -1,30 +1,34 @@
...
-.PHONY: docs/install/markdown
-docs/install/markdown: DESTDIR:=$(DESTDIR)/markdown
-docs/install/markdown: ## Generate CLI reference in markdown format
-	@DESTDIR=$(DESTDIR) FORMAT=markdown go run $(TOOLS_DIR)/docs/generate.go
-
-.PHONY: docs/install/manpages
-docs/install/manpages: DESTDIR:=$(DESTDIR)/manpages
-docs/install/manpages: ## Generate CLI reference in man(1) format
-	@DESTDIR=$(DESTDIR) FORMAT=man go run $(TOOLS_DIR)/docs/generate.go
+DOCS_CMD_FORMAT=markdown
```
Comment (bartsmykla) [inline mk/docs.mk:15]: I know we need markdown here, but maybe for futureproofing we could go with `?=` 
```suggestion
DOCS_CMD_FORMAT ?= markdown
```

**Review by bartsmykla — APPROVED** [review]
lgtm, just few nits

**Pull request merged**